src/server/scripts/Northrend/Naxxramas/boss_razuvious.cpp

```diff
@@ -171,6 +285,9 @@ class boss_razuvious : public CreatureScript
             }
             DoMeleeAttackIfReady();
         }
+
+    private:
+        ObjectGuid _rpBuddyGUID;
```

Comment:
The following exists for Map `instance->GetCreature(someGUID)`

https://github.com/azerothcore/azerothcore-wotlk/blob/0a8175ef037732aa25a908687a2c3a019ba5eb55/src/server/game/Maps/Map.cpp#L3314

however from BossAI `instance->GetCreature(uint32 type)` is called. Lookup is done by "type". `DATA_` set in minionData or creatureData. Those structs are not implemented for naxx

https://github.com/azerothcore/azerothcore-wotlk/blob/0a8175ef037732aa25a908687a2c3a019ba5eb55/src/server/game/Instances/InstanceScript.cpp#L99

https://github.com/azerothcore/azerothcore-wotlk/blob/0a8175ef037732aa25a908687a2c3a019ba5eb55/src/server/game/Instances/InstanceScript.cpp#L82

```
MinionData const minionData[] =
{
    { NPC_DK_UNDERSTUDY,  DATA_RAZUVIOUS },
```
Even with these implemented, I think this stores every single DK under minionData? Not sure how you can store/find the selected rpBuddy this way

`ObjectAccessor::GetCreature` can be replaced with `me->GetMap()->GetCreature(_rpBuddyGUID)`, but I see `ObjectAccessor::GetCreature` is more common

Reply:
@Nyeriah You are script pro. Help us out :D

Reply:
The modern storage method only works with single spawns unfortunately, because of indexation (single creature objects indexed by the DATA value we choose)